### PR TITLE
[DTensor] Add rand_like, randn_like, randint_like ops to shard propagation

### PR DESCRIPTION
**Pull request by joshlk**
Add rand_like, randn_like, randint_like ops to shard propagation

cc: @wanchaol

### CODE REVIEW

File: torch/distributed/_tensor/ops/tensor_ops.py [hunk 1]
```diff
@@ -64,6 +64,11 @@ def default_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> StrategyType:
         aten.empty_like.default,
         aten.full_like.default,
         aten.ones_like.default,
+        aten.rand_like.default,
+        aten.randn_like.default,
+        aten.randint_like.default,
+        aten.randint_like.low_dtype,
+        aten.randint_like.low_dtype_out,
```
Comment (wanchaol) [inline torch/distributed/_tensor/ops/tensor_ops.py:71]: I think we should add RuntimeSchemaInfo that includes "dtype" kwarg so that sharding propagation cache different dtype configs correctly. 

like this https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/ops/tensor_ops.py#L109

File: test/distributed/_tensor/test_random_ops.py [hunk 2]
```diff
@@ -72,6 +72,9 @@ def test_init_ops(self):
         )
         self._run_init_op(torch.nn.init.normal_, mean=1.5, std=0.8)
         self._run_init_op(torch.nn.init.uniform_, a=0, b=1.2)
+        self._run_init_op(torch.rand_like)
+        self._run_init_op(torch.randn_like)
+        self._run_init_op(torch.randint_like, low=0, high=100)
```
Comment (wanchaol) [inline test/distributed/_tensor/test_random_ops.py:77]: nit: maybe also add some tests about different dtypes

**Review by wanchaol — APPROVED** [review]
lgtm, some suggestions about dtype

**Review by fduwjj — APPROVED** [review]
I added the label for you, you can remerge it, it should be super fast.
